src/functions-and-arrays.js

```diff
@@ -1,73 +1,148 @@
-  [21, 36, 23, 9, 75, 0, 76, 44, 20, 45, 35, 14, 0, 61, 33, 97, 34, 31, 33, 95],
-  [78, 17, 53, 28, 22, 75, 31, 67, 15, 94, 3, 80, 4, 62, 16, 14, 9, 53, 56, 92],
-  [16, 39, 5, 42, 96, 35, 31, 47, 55, 58, 88, 24, 0, 17, 54, 24, 36, 29, 85, 57],
-  [86, 56, 0, 48, 35, 71, 89, 7, 5, 44, 44, 37, 44, 60, 21, 58, 51, 54, 17, 58],
-  [19, 80, 81, 68, 5, 94, 47, 69, 28, 73, 92, 13, 86, 52, 17, 77, 4, 89, 55, 40],
-  [4, 52, 8, 83, 97, 35, 99, 16, 7, 97, 57, 32, 16, 26, 26, 79, 33, 27, 98, 66],
-  [88, 36, 68, 87, 57, 62, 20, 72, 3, 46, 33, 67, 46, 55, 12, 32, 63, 93, 53, 69],
-  [4, 42, 16, 73, 38, 25, 39, 11, 24, 94, 72, 18, 8, 46, 29, 32, 40, 62, 76, 36],
-  [20, 69, 36, 41, 72, 30, 23, 88, 34, 62, 99, 69, 82, 67, 59, 85, 74, 4, 36, 16],
-  [20, 73, 35, 29, 78, 31, 90, 1, 74, 31, 49, 71, 48, 86, 81, 16, 23, 57, 5, 54],
-  [1, 70, 54, 71, 83, 51, 54, 69, 16, 92, 33, 48, 61, 43, 52, 1, 89, 19, 67, 48]
-];
+function maxOfTwoNumbers(firstNum, secondNum){
+  if (firstNum > secondNum)
+  return (firstNum)
+  else if (firstNum < secondNum)
+  return (secondNum)
+  else if (firstNum === secondNum)
+  return (firstNum)
+  }
+  
+  
+  // Iteration #2: Find longest word
+  const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
+  
+  function findLongestWord(words){
+    let lword = "";
+  
+    if (words.length === 0) {return null;} 
+    else if (words.length === 1) {
+      return words[0];
+    }
+  
+    for (let i = 0; i < words.length; i++) {
+      words[i];
+      if (lword.length < words[i].length) {
+        lword = words[i];
+      }
+    } 
+  return lword;
+  }
+  
+  // Iteration #3: Calculate the sum
+  
+  const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+  function sumNumbers(number) {
+    let result = 0;
+    for (let i = 0; i < number.length; i++) {
+        result = result + number[i];
+    }
+    return result; 
+  
+  }
+   function sum (numbers){
+    if(numbers.length === 0) {
+      return 0;
+    } else {
+      let result = 0;
+    for (let i = 0; i < number.length; i++) {
+        result = result + number[i];
+    }
+    return result; 
+    }
+  
+   }
+  
+  
+  // Iteration #4: Calculate the average
+  // Level 1: Array of numbers
+  const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
+  function averageNumbers(numbersAvg) {
+    if (numbersAvg.length <= 0) {return null;} 
+  }
+  
+  // Level 2: Array of strings
+  const wordsArr = ['seat', 'correspond', 'linen', 'motif', 'hole', 'smell', 'smart', 'chaos', 'fuel', 'palace'];
+  function averageWordLength(words){
+    if (words.length <= 0 ) {return null;}
+    
+  }
+  function avg (words){
+    if (words.length <= 0 ) {return null;}
+  }
+  
+  
+  // Iteration #5: Unique arrays
+  const wordsUnique = [
+    'crab',
+    'poison',
+    'contagious',
+    'simple',
+    'bring',
+    'sharp',
+    'playground',
+    'poison',
+    'communion',
+    'simple',
+    'bring'
+  ];
+  function uniquifyArray (words) {
```

Comment:
Este era complicadete. Se puede hacer con un for y un filter por ejemplo. De todas formas la búsqueda de "how to remove duplicates from array js" arroja varias soluciones.